### PR TITLE
[dotnet] Fix `WebDriver.AuthenticatorId`

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

This fixes the `WebDriver.AuthenticatorId` ID to reflect the currently active authenticator ID.

### Motivation and Context

The property is currently completely broken, returning `null` exclusively. Another victim of the non-auto property typo.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Fixed the `WebDriver.AuthenticatorId` to correctly reflect the active authenticator ID by replacing the field with a property.
- Updated all relevant methods in `WebDriver` to use the new `AuthenticatorId` property.
- Added a test in `VirtualAuthenticatorTest` to verify that `AuthenticatorId` is set correctly and is null after removal.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>WebDriver.cs</strong><dd><code>Fix and refactor `AuthenticatorId` handling in WebDriver</code>&nbsp; </dd></summary>
<hr>

dotnet/src/webdriver/WebDriver.cs

<li>Replaced <code>authenticatorId</code> field with a property <code>AuthenticatorId</code>.<br> <li> Updated methods to use the <code>AuthenticatorId</code> property.<br> <li> Ensured <code>AuthenticatorId</code> is set and retrieved correctly.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14814/files#diff-ef1874c09b1dca0d62ea1f0ead685355944b9cb7ae63df60ad9b32abdf90c8dc">+10/-11</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>VirtualAuthenticatorTest.cs</strong><dd><code>Add test for `AuthenticatorId` functionality in WebDriver</code></dd></summary>
<hr>

dotnet/test/common/VirtualAuthn/VirtualAuthenticatorTest.cs

<li>Added assertion to verify <code>AuthenticatorId</code> is set correctly.<br> <li> Ensured test checks for correct removal of authenticator.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14814/files#diff-914e73dc63b78b107b4030af3736fedda7e97a27008b22e9a73eb64227865a16">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by RenderMichael** [comment]
Both AI suggestions are completely correct

1. `RemoveVirtualAuthenticator` does not even use its parameter
2. There's no validation to make sure a virtual authenticator is in place (unhelpful exception message)

I want to fix these, but I don't want this to become a singular "improve virtual authenticator" PR. This PR fixes a bug, and a follow-up PR can improve anything else that could use improvement.

**Review by nvborisenko — APPROVED** [review]
Thanks, nice catch!

**Comment by nvborisenko** [comment]
`//dotnet/test/common:VirtualAuthn/VirtualAuthenticatorTest-chrome` is failing, please take a look.

**Comment by RenderMichael** [comment]
Funny. The one test I changed for this, is the only one passing.

image

This is because the test teardown removes the VA, an operation which is guarded by `webDriver.AuthenticatorId != null` and only now has begun to work. Let me fix this.

**Comment by RenderMichael** [comment]
Fixed

image

File: dotnet/src/webdriver/WebDriver.cs [hunk 1]
```diff
@@ -1057,15 +1056,15 @@ public string AddVirtualAuthenticator(VirtualAuthenticatorOptions options)
...
             this.Execute(DriverCommand.RemoveVirtualAuthenticator, parameters);
-            this.authenticatorId = null;
+            this.AuthenticatorId = null;
         }
 
         /// <summary>
         /// Gets the virtual authenticator ID for this WebDriver instance.
         /// </summary>
-        public string AuthenticatorId { get; }
+        public string AuthenticatorId { get; private set; }
```
Comment (nvborisenko) [inline dotnet/src/webdriver/WebDriver.cs:1067]: Then it might be nullable? Contribution to NRT?

Comment (RenderMichael) [reply dotnet/src/webdriver/WebDriver.cs:1067]: `WebDriver` is a very large class. It feels awkward to add `#nullable enable` before this property and `#nullable restore` right afterwards. It seems more complicated to do that, than to add `?` when I annotate this class.

Comment (nvborisenko) [reply dotnet/src/webdriver/WebDriver.cs:1067]: If you preferer to make it later, np. At least here we understand it really might be null.

**Pull request merged**